buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -306,9 +321,33 @@ public void plugin(String pluginProjectPath) {
     @Override
     public void module(Provider<RegularFile> module) {
         checkFrozen();
+        registerExtractedConfig(module);
         this.modules.add(module.map(RegularFile::getAsFile));
     }
 
+    private void registerExtractedConfig(Provider<RegularFile> pluginProvider) {
```

Comment:
Nice. So this avoids us having to do a temporary unpack on all of these every time just to do input snapshotting, right?

Reply:
yes, as long as the input doesn't change this shouldn't need to be unpacked again. This currently works only with up-to-date checks. I'll look into making these transforms using the build cache in a next step

Reply:
I'd be curious if there's a significant performance benefit there. My suspicion is probably not since this basically all just I/O. At least locally you only have to do it once.